src/main/java/org/jitsi/videobridge/SctpConnection.java

```diff
@@ -382,41 +411,51 @@ public boolean expire()
     }
 
     /**
-     * Returns default <tt>WebRtcDataStream</tt> if it's ready or <tt>null</tt>
-     * otherwise.
-     * @return <tt>WebRtcDataStream</tt> if it's ready or <tt>null</tt>
-     *         otherwise.
-     * @throws IOException
+     * Select a WebRTC data channel that should be used for writing by default.
+     *
+     * The strategy is try to select client's data channel with the highest SID
+     * number first. A data channel with the highest SID will usually be the
+     * most recent one, because the WebRTC stack tries to allocate SIDs in
+     * order.
+     *
+     * The distinction between client's opened channels vs JVB opened channels
+     * comes from the fact that the RFC requires the DTLS initiator to use even
+     * SID numbers and the responder to use odd numbers.
+     *
+     * @return a writeable <tt>WebRtcDataStream</tt> if it's ready or
+     * <tt>null</tt> otherwise.
      */
     public WebRtcDataStream getDefaultDataStream()
-        throws IOException
     {
-        WebRtcDataStream def;
-
         synchronized (syncRoot)
         {
-            if (sctpSocket == null)
-            {
-                def = null;
-            }
-            else
+            if (sctpSocket != null)
             {
-                // Channel that runs on sid 0
-                def = channels.get(0);
-                if (def == null)
-                    def = openChannel(0, 0, 0, 0, "default");
-
-                // Pawel Domas: Must be acknowledged before use.
-                // XXX Lyubomir Marinov: We're always sending ordered. According
-                // to "WebRTC Data Channel Establishment Protocol", we can start
-                // sending messages containing user data after the
-                // DATA_CHANNEL_OPEN message has been sent without waiting for
-                // the reception of the corresponding DATA_CHANNEL_ACK message.
-//                if (!def.isAcknowledged())
-//                    def = null;
+                // Tries to elect a stream with the highest client SID
+                WebRtcDataStream highestClientSid
+                    = channels.values()
+                        .stream()
+                        .filter(
+                            s -> isInitiator()
+                                ? s.getSid() % 2 == 1
+                                : s.getSid() % 2 == 0)
+                        .reduce((s1, s2) -> s1.getSid() > s2.getSid() ? s1 : s2)
```

Comment:
@paweldomas looks like [`.max(Comparator.comparingInt(x -> x.getSid())`](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#comparingInt-java.util.function.ToIntFunction-) would be more explicit way to express lookup channel with biggest sid.

Reply:
yeah, thanks ! that looks useful